### ISSUE TITLE
Some doubts about gradient when train GAN

### ISSUE THREAD

**Issue by UmiUmiU2333** [ginobilinie/asdnet#2]
Hi, I have some doubts about step.5.2 update D network in line 924 of runGAN.py.

If we run loss.backward() in line 942 and line 953, and run optimizer.step() in line1156 and line1157, then gradients computed by line 942 and line 953 may be used to update G & D, while G should not be updated here in general.

**Comment by ginobilinie** [comment]
@UmiUmiU2333 Theoretically, you're right about this point. However, it doesnot matter much in practice.
